src/mds/MDCache.cc

```diff
@@ -7627,6 +7627,12 @@ int MDCache::path_traverse(MDRequestRef& mdr, Message *req, MDSInternalContextBa
   if (cur->state_test(CInode::STATE_PURGING))
     return -ESTALE;
 
+  // make sure snaprealm are open...
+  if (mdr && cur->snaprealm && !cur->snaprealm->open &&
+      !cur->snaprealm->open_parents(_get_waiter(mdr, req, fin))) {
```

Comment:
Do we really not have a more abstracted way to check this than by reaching inside the SnapRealm's data members directly?
That sort of thing is common in our MDS code but definitely ripe for improvement...